core/src/vaults.ts

```diff
@@ -381,6 +389,41 @@ const getLatestVault = async (network: string, walletAddress: string) => {
     return new BigNumber(lastHex._hex).toNumber();
 };
 
+export const openVaultWithProxiedContractAndDrawDebt = async (
+    network: string,
+    proxyAddress: string,
```

Comment:
Can you open a vault with proxied contract without creating a proxy upfront? Then should `createProxy` be moved into this function and not used outside? (so that we get clear API of this function)

Reply:
No and no.

This would force creation of the new proxy each time when the vault is opened by our functionality.
This is undesired because single DSProxy instance is sufficient (reusable), can be used for variety of purposes and not only for the current action we're performing.

Reply:
> This would force creation of the new proxy each time when the vault is opened by our functionality.

It wouldn't if it will be optional, eg `proxyAddress?: string`. Anyway, please see my other comment about proxy creation https://github.com/sidestream-tech/unified-auctions-ui/pull/510#discussion_r1006737780

Reply:
i don't see the point. As this function is the part of `src` (not specifically dedicated to simulations), i would have to instead provide the proxyOwner address - then there're 2 mutually exclusive arguments. 

Should i then move this func into simulation suite for the specific use exclusively inside it?